pkg/distributor/distributor.go

```diff
@@ -1149,7 +1207,8 @@ func (d *Distributor) prePushHaDedupeMiddleware(next PushFunc) PushFunc {
 		}
 
 		haReplicaLabel := d.limits.HAReplicaLabel(userID)
-		cluster, replica := findHALabels(haReplicaLabel, d.limits.HAClusterLabel(userID), req.Timeseries[0].Labels)
+		haClusterLabel := d.limits.HAClusterLabel(userID)
+		cluster, replica := findHALabels(haReplicaLabel, haClusterLabel, req.Timeseries[0].Labels)
```

Comment:
i think it's a bit confusing that we log only the first replicas even though we'll now ingest data from multiple replicas. can we add the tracing instrumentation further down as we're looping over the replica states?